src/emqx_cm.erl

```diff
@@ -142,11 +142,14 @@ insert_channel_info(ClientId, Info, Stats) ->
 %% Note that: It should be called on a lock transaction
 register_channel(ClientId, ChanPid, #{conn_mod := ConnMod}) when is_pid(ChanPid) ->
     Chan = {ClientId, ChanPid},
+    %% Note that cast/1 should be called before ets:insert/2 to reduce the possibility
+    %% of leakage in ETS tables due to the process dies before being monitored.
+    cast({registered, Chan}),
```

Comment:
do cast here still cannot ensure the ordering since it is async but it is by far the best effort we could do here.

Reply:
I cannot think off a reason why there is still a risk of leak.
the casted pid will eventually get monitored, and immediately triggers a DOWN which in turn leads to the ets table deletion.